kubernetes/kubernetes.go

```diff
@@ -123,6 +123,11 @@ const rbacError string = `RBAC role in the Cloudability namespace may need to be
 // CollectKubeMetrics Collects metrics from Kubernetes on a predetermined interval
 // nolint: gocyclo
 func CollectKubeMetrics(config KubeAgentConfig) {
+	isInvalidClusterName, _ := regexp.MatchString(`[^a-zA-Z\d-]`, config.ClusterName)
+	if isInvalidClusterName {
+		log.Fatalf("Invalid agent configuration. CLOUDABILITY_CLUSTER_NAME must contain only alphanumeric characters and dashes "+
+			"and is currently set to %s.", config.ClusterName)
+	}
```

Comment:
Would maybe be more appropriate in `CheckRequiredSettings`

Reply:
Moved all the logic over 👍